pkg/operator/operator.go

```diff
@@ -133,51 +139,45 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		return err
 	}
 
-	availableCnd := operatorapi.OperatorCondition{
-		Type:   controllerName + operatorapi.OperatorStatusTypeAvailable,
-		Status: operatorapi.ConditionTrue,
+	if !platformSupported {
```

Comment:
Previously, sync set `Available=true` on an supported platform, now it does not set anything. I know it's just a theoretic case

Reply:
Yep. I think it should be okay.